test/jsonValidationTest.js

```diff
@@ -5,120 +5,167 @@ const fs = require('fs');
+    it('Verifying workbook settings.json files', function (done) {
         browseDirectory(cohortPath, (error, results) => {
             if (error) throw error;
             results.filter(file => file.substr(-13) === 'settings.json')
                 .forEach(file => {
-                    console.log(file);
-                    validateJsonString(file);
+                    let json = validateJsonStringAndGetObject(file)
+                    validateSettingsForWorkbook(json, file);
                 });
-            
-                done();
+
+            done();
         });
     });
 
-    it('Verifying workbook category json files', function(done){
+    it('Verifying workbook category json files', function (done) {
         browseDirectory(cohortPath, (error, results) => {
             if (error) throw error;
             results.filter(file => file.substr(-22) === 'categoryResources.json')
                 .forEach(file => {
-                    validateJsonString(file);
+                    let json = validateJsonStringAndGetObject(file);
+                    validateCategory(json, file);
                 });
-            
-                done();
+
+            done();
         });
     });
 });
 
-function validateJsonString(file) {
-    let json = fs.readFileSync(file, 'utf8');
-    assert.isTrue(isJsonString(json), 'Invalid json format with \'' + file + '\'');
-}
-
-var browseDirectory = function(dir, done) {
+var browseDirectory = function (dir, done) {
     var results = [];
-    fs.readdir(dir, function(err, list) {
-      if (err) return done(err);
-      var i = 0;
-      (function next() {
-        var file = list[i++];
-        if (!file) return done(null, results);
-        file = dir + '/' + file;
-        fs.stat(file, function(err, stat) {
-          if (stat && stat.isDirectory()) {
-            browseDirectory(file, function(err, res) {
-              results = results.concat(res);
-              next();
+    fs.readdir(dir, function (err, list) {
+        if (err) return done(err);
+        var i = 0;
+        (function next() {
+            var file = list[i++];
+            if (!file) return done(null, results);
+            file = dir + '/' + file;
+            fs.stat(file, function (err, stat) {
+                if (stat && stat.isDirectory()) {
+                    browseDirectory(file, function (err, res) {
+                        results = results.concat(res);
+                        next();
+                    });
+                } else {
+                    results.push(file);
+                    next();
+                }
             });
-          } else {
-            results.push(file);
-            next();
-          }
-        });
-      })();
+        })();
     });
-  };
+};
 
+function validateJsonStringAndGetObject(file) {
+    let json = fs.readFileSync(file, 'utf8');
+    let obj = TryParseJson(json)
+    assert.isTrue(obj != null, 'Invalid json format with \'' + file + '\'');
```

Comment:
why not catch the error and actually output it in the assert?  if the file is invalid, why not explain why right here?

Reply:
Good suggestion